### PR TITLE
chore(dev): Transition local k8s dev to Tilt

### LINKED ISSUE #11060: scripts/skaffold.sh essentially unusable on MacOS
The local kubernetes dev workflow is more or less broken on Mac's, it seems to take hours to even build the vector image and on top of that we seem to possibly be sending the entire contents of `target/` to the docket daemon for the build context

### PR DESCRIPTION
**Pull request by spencergilbert**
Closes #11060

Running `tilt up` will build and deploy to your local cluster (`tilt` being available through a number of package managers)

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+###################
```
Comment (spencergilbert) [inline Tiltfile:1]: These seem to map 1:1 (more or less) to the CLI equivalent of docker or helm, so I didn't document them here.

File: tilt/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+ARG RUST_VERSION
```
Comment (spencergilbert) [inline tilt/Dockerfile:1]: Aside from the configurable FEATURES, I essentially stole this from the soak/Dockerfile and soak/Dockerfile.build

File: Tiltfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# VECTOR TILTFILE #
+###################
+
+load('ext://helm_resource', 'helm_resource', 'helm_repo')
+
+docker_build(
+    ref='timberio/vector',
+    context='.',
+    build_args={'RUST_VERSION': '1.58'},
```
Comment (spencergilbert) [inline Tiltfile:10]: I'm not super happy with this but I didn't see a way to pull it from the toolchain file, probably is a way though

**Comment by spencergilbert** [comment]
The build steps took ~5m on my M1 MacBook Pro

**Review by jszwedko — APPROVED** [review]
Nice! I like it. I was able to run `tilt up` and get everything setup.

Do you think we should continue to maintain the skaffold files or just drop those now? I'd prefer to only support one common development workflow.

**Comment by spencergilbert** [comment]
> Nice! I like it. I was able to run `tilt up` and get everything setup.
> 
> Do you think we should continue to maintain the skaffold files or just drop those now? I'd prefer to only support one common development workflow.

I'm pro dropping skaffold completely - but the e2e tests are still using some bits from there (currently). I'd like to finalize that and make sure `tilt` is working happily on intel and linux too.

**Comment by jszwedko** [comment]
> > Nice! I like it. I was able to run `tilt up` and get everything setup.
> > Do you think we should continue to maintain the skaffold files or just drop those now? I'd prefer to only support one common development workflow.
> 
> I'm pro dropping skaffold completely - but the e2e tests are still using some bits from there (currently). I'd like to finalize that and make sure `tilt` is working happily on intel and linux too.

Sounds good 👍

**Pull request merged**